rbd-target-api.py

```diff
@@ -538,6 +545,97 @@ def _gateway(gateway_name=None):
             return jsonify(message="Gateway removed successfully"), 200
 
 
+def define_luns(gateway):
+    """
+    define the disks in the config to LIO
+    :param gateway: (object) gateway object - used for mapping
+    :return: None
+    """
+
+    local_gw = this_host()
+
+    # sort the disks dict keys, so the disks are registered in a specific
+    # sequence
+    disks = config.config['disks']
+    srtd_disks = sorted(disks)
+    pools = {disks[disk_key]['pool'] for disk_key in srtd_disks}
+
+    if pools:
+        with rados.Rados(conffile=settings.config.cephconf) as cluster:
```

Comment:
Let's use the same coding style as the rest of the code instead of doing with x as y: just do y = x.

We should also have a lib/ceph-iscsi-config function to do this work and then we also do not have code like this having to know details like the cephconf settings type of details. Maybe just some RBDDev static function lookup() that takes a image and pool name and will return a RBDDev or maybe a modified constructor that looks up the object and fills in values like size if the device already exists. You would then do
```

for disk in config.config['disks']:
   pool = disk['pool']
   image_name = disk['image']
   rbd_image = RBDDev.lookup(pool, image_name)
    lun = LUN(logger, pool, image_name, rbd_image.size, disk['owner])
    lun.allocate()
```

Reply:
I think the purpose of using pool is to reduce the connection times with cluster. The code is copied from ceph-iscsi-config/rbd-target-gw.py. I have tried to rewrite the patch with your suggestion, but if I want to make the code structure better, I have to reduce the performance. 
@mikechristie @pcuzner  How do you think?

Reply:
Nice, you are right.

Let's just move the define_luns and define_client functions from rbd-target-gw to ceph-iscsi-config so we can share them.